CHANGELOG.md

```diff
@@ -1,5 +1,54 @@
+  * All values are passed through to notification listeners.
+  * Strings, booleans, and valid numbers are passed to the event dispatcher and can be used for Optimizely results segmentation. A valid number is a finite number in the inclusive range [-2⁵³, 2⁵³]. 
+  * Strings, booleans, and valid numbers are relevant for audience conditions.
+* Support for additional matchers in audience conditions ([#337](https://github.com/optimizely/objective-c-sdk/pull/337)):
+  * An `exists` matcher that passes if the user has a non-null value for the targeted user attribute and fails otherwise.
+  * A `substring` matcher that resolves if the user has a string value for the targeted attribute.
+  * `gt` (greater than) and `lt` (less than) matchers that resolve if the user has a valid number value for the targeted attribute. A valid number is a finite number in the inclusive range [-2⁵³, 2⁵³].
+  * The original (`exact`) matcher can now be used to target booleans and valid numbers, not just strings.
+* Support for A/B tests, feature tests, and feature rollouts whose audiences are combined using `"and"` and `"not"` operators, not just the `"or"` operator.
+* Datafile-version compatibility check: The SDK will remain uninitialized (i.e., will gracefully fail to activate experiments and features) if given a datafile version greater than 4.
+* Updated Pull Request template and commit message guidelines.
+* Support for empty bucketing IDs when evaluating feature rollouts ([#367](https://github.com/optimizely/objective-c-sdk/pull/367)).
+
+### Breaking Changes:
+* Previously, notification listeners were only given string-valued user attributes because only strings could be passed into various method calls. That is no longer the case. The `ActivateListener` and `TrackListener` interfaces now receive user attributes as `NSDictionary<NSString *, NSObject *> * _Nullable` instead of  `NSDictionary<NSString *,NSString *> * _Nonnull`.
+* The Localytics integration now relies on the user profile service to attribute events to the user's variations.
+* Drops support for the `variableBoolean`, `variableDouble`, `variableInteger`, and `variableString` APIs. Please migrate to the Feature Management APIs which were introduced in the 2.x releases.
+
+### Bug Fixes:
+* Experiments and features can no longer activate when a negatively targeted attribute has a missing, null, or malformed value.
+  * Audience conditions (except for the new `exists` matcher) no longer resolve to `false` when they fail to find an legitimate value for the targeted user attribute. The result remains `null` (unknown). Therefore, an audience that negates such a condition (using the `"not"` operator) can no longer resolve to `true` unless there is an unrelated branch in the condition tree that itself resolves to `true` ([#355](https://github.com/optimizely/objective-c-sdk/pull/355)). 
```

Comment:
Maybe
```suggestion
  * Audience conditions (except for the new `exists` matcher) no longer resolve to `false` when they fail to find an legitimate value for the targeted user attribute. The result remains `null` (unknown). Therefore, an audience that negates such a condition (using the `"not"` operator) can no longer resolve to `true` unless there is an unrelated branch in the condition tree that itself resolves to `true` ([#309](https://github.com/optimizely/objective-c-sdk/pull/309), [#355](https://github.com/optimizely/objective-c-sdk/pull/355)).
```
since some of the underlying functionality was introduced earlier

Though I'm not too particular about this!